### PR TITLE
Accept source_rev and target_rev attributes in comments

### PR DESCRIPTION
**Pull request by hellcp-work**
Stores additional info in the database to display the correct diff for when the comment was created.

### CODE REVIEW

**Comment by hennevogel** [comment]
I'm not sure why you stuff all of this into `diff_ref`. Why don't we add  `Comment.source_rev` and `Comment.target_rev`?

**Comment by hellcp-work** [comment]
> I'm not sure why you stuff all of this into `diff_ref` and parse it out afterward. Why don't we add `Comment.source_rev` and `Comment.target_rev`?

That might be an even better idea

**Comment by hennevogel** [comment]
> That might be an even better idea

Also a good opportunity to rename this to `diff_line` or something I would say...

**Comment by eduardoj** [comment]
> Also a good opportunity to rename this to `diff_line` or something I would say...

I'll take care of renaming this column to `diff_line` in a series of pull requests based on this one. Renaming a column is not a safe operation: https://github.com/ankane/strong_migrations?tab=readme-ov-file#renaming-a-column

File: src/api/app/components/diff_component.html.haml [hunk 1]
```diff
@@ -41,15 +41,15 @@
         - if commentable
           .line-comment{ id: "comment#{id}" }
             - if commented_lines.include?(id)
-              - commentable.comments.where(diff_ref: id).each do |comment|
+              - commentable.comments.where(diff_ref: id, source_rev: [source_rev, nil], target_rev: [target_rev, nil]).each do |comment|
```
Comment (hennevogel) [inline src/api/app/components/diff_component.html.haml:44]: What else is there?

**Comment by eduardoj** [comment]
> > Also a good opportunity to rename this to `diff_line` or something I would say...
> 
> I'll take care of renaming this column to `diff_line` in a series of pull requests based on this one. Renaming a column is not a safe operation: https://github.com/ankane/strong_migrations?tab=readme-ov-file#renaming-a-column

`diff_ref` has the following format: `"diff_#{file_index}_n#{line.index}"`. For example: `diff_4_n23`. See: https://github.com/openSUSE/open-build-service/blob/db804298c1162588ce262e586dbaec103bfd01ca/src/api/app/components/diff_component.html.haml#L4

I find renaming `diff_ref` to `diff_line` to be less accurate. It misses the "file index" part while keeping the "line", introducing some confusion.

I also find the alternatives `diff_file_line`, `diff_fileindex_line` or `diff_fileindex_linenumber` very long.

I would prefer to leave the name of the variable like it is now. Do you have another alternative which could fit, @hennevogel?

**Comment by hennevogel** [comment]
> diff_fileindex_linenumber

Sounds good to me. I don't care how long it is...

**Comment by eduardoj** [comment]
> > I would prefer to leave the name of the variable like it is now. Do you have another alternative which could fit, @hennevogel?
> 
> There's also the alternative of splitting it into two columns, since that would make parsing unnecessary

I find this option better, say `diff_file_index` and `diff_line_number`. On it.

File: src/api/db/schema.rb [hunk 2]
```diff
@@ -373,8 +373,13 @@
     t.string "diff_ref"
     t.datetime "moderated_at"
     t.integer "moderator_id"
+<<<<<<< HEAD
```
Comment (hennevogel) [inline src/api/db/schema.rb:376]: Merge conflicts...

**Review by hennevogel — APPROVED** [review]

**Review by eduardoj — APPROVED** [review]

**Pull request merged**